docs/fundamentals/code-analysis/quality-rules/ca2255.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "CA2255: The `ModuleInitializer` attribute should not be used in libraries"
+description: "Learn about code analysis rule CA2255: The `ModuleInitializer` attribute should not be used in libraries"
+ms.date: 11/15/2021
+ms.topic: reference
+f1_keywords:
+- CA2255
+- ModuleInitializerAttributeShouldNotBeUsedInLibraries
+helpviewer_keywords:
+- ModuleInitializerAttributeShouldNotBeUsedInLibraries
+- CA2255
+author: jeffhandley
+---
+# CA2255: The `ModuleInitializer` attribute should not be used in libraries
+
+| | Value |
+|-|-|
+| **Rule ID** |CA2255|
+| **Category** |[Usage](usage-warnings.md)|
+| **Fix is breaking or non-breaking** |Breaking|
```

Comment:
I do not remember when exactly we could say it is breaking, though don't think it is breaking

Reply:
I wasn't sure here either. There's no code fix supplied, but it's a "breaking design change" that the developer would need to make to react to the diagnostic. I'll switch to non-breaking though.